managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpAddEntry.java

```diff
@@ -110,6 +125,13 @@ public void failed(ManagedLedgerException e) {
 
     @Override
     public void addComplete(int rc, final LedgerHandle lh, long entryId, Object ctx) {
+
+        if (!STATE_UPDATER.compareAndSet(OpAddEntry.this, State.INITIATED, State.COMPLETED)) {
+            log.warn("[{}] The add op is terminal legacy callback for entry {}-{} adding.", ml.getName(), lh.getId(), entryId);
+            OpAddEntry.this.recycle();
```

Comment:
since we are creating a new entry when retrying the ops on the new ledger, do we need to introduce the `state` field and recycle here?

Since this op is only used by the old ledger handler, it will not be reused across ledgers. It should already be recycled correctly.